### Com a EDA feita, agora será feito testes estatísticos para maior rigor nas informações observadas nos dados

Para isso, usaremos a biblioteca Scipy


In [30]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import scipy

In [26]:
df = pd.read_csv('../dados/dados_tratados/prf_tratado.csv')
df.head()

,id,data_inversa,dia_semana,horario,uf,br,km,municipio,causa_acidente,tipo_acidente,...,feridos_graves,ilesos,ignorados,feridos,veiculos,latitude,longitude,feriado,turno,acidente_fatal
0,331730.0,2021-01-01,sexta-feira,05:30:00,SP,116,453,REGISTRO,Reação tardia ou ineficiente do condutor,Tombamento,...,0,0,0,1,1,-24.561681,-47.867522,True,Madrugada,0
1,331804.0,2021-01-01,sexta-feira,08:05:00,AM,174,937,MANAUS,Reação tardia ou ineficiente do condutor,Colisão traseira,...,0,5,1,1,5,-2.508068,-60.036434,True,Manhã,0
2,331815.0,2021-01-01,sexta-feira,10:10:00,CE,222,"128,5",ITAPAJE,Velocidade Incompatível,Tombamento,...,0,0,11,2,12,-3.707626,-39.623509,True,Manhã,0
3,331823.0,2021-01-01,sexta-feira,12:30:00,RJ,493,18,MAGE,Ingestão de álcool e/ou substâncias psicoativa...,Colisão lateral,...,0,1,1,1,3,-22.668712,-43.018627,True,Tarde,0
4,331843.0,2021-01-01,sexta-feira,14:40:00,RJ,393,252,BARRA DO PIRAI,Condutor Dormindo,Colisão frontal,...,1,1,1,2,3,-22.434220,-43.775615,True,Tarde,0


O dia da semana influencia a fatalidade de um acidente?

Hipótese-nula: Todos os dias são igualmente fatais <br>
Hipótese alternativa: O dia da semana influencia a fatalidade

p-value < 5%

In [ ]:
#Criar tabela de contigência

tab_cont_dia = pd.crosstab(df['acidente_fatal'],df['dia_semana'])
tab_cont_dia

dia_semana,domingo,quarta-feira,quinta-feira,segunda-feira,sexta-feira,sábado,terça-feira
acidente_fatal,,,,,,,
0,51410,40581,41365,44099,49401,51573,39571
1,4868,2810,2977,3108,3607,4538,2711


In [ ]:
#MEDIR CHI²

chi2, p, gl, esperado = scipy.stats.chi2_contingency(tab_cont_dia)

print(f'χ² = {chi2:.2f}')
print(f'Valor-p = {p:.4e}')
print(f'Graus de liberdade = {gl}')

χ² = 372.09
Valor-p = 2.7815e-77
Graus de liberdade = 6


In [ ]:
#DESCOBRIR O V de Cramér

n = tab_cont_dia.sum().sum()
v_cramer = np.sqrt(chi2 / (n * (min(tab_cont_dia.shape) - 1)))
print(f"V de Cramér = {v_cramer:.3f}")

V de Cramér = 0.033


Com o valor-p de 2.78×10⁻⁷², podemos, com confiança, negar a hipótese nula e concluir que o dia da semana afeta a taxa de fatalidade.

Porém, com o V de Cramér em 0.033, a associação é considerada de magnitude fraca, algo que é reforçado com a taxa de fatalidade por dia da semana visto anteriormente